File: src/Person.java
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlType;

@XmlType(name="person", namespace="http://statmod.ru/staff")
public class Person {
    @XmlAttribute(name="id")
    private String id;

    @XmlElement(name="fn", namespace="http://statmod.ru/staff")
    private String firstName;

    @XmlElement(name="mn", namespace="http://statmod.ru/staff")
    private String middleName;

    @XmlElement(name="ln", namespace="http://statmod.ru/staff")
    private String lastName;

    /** Имя */
    public String getFirstName() {
        return firstName;
    }

    /** Фамилия */
    public String getLastName() {
        return lastName;
    }

    /** Отчество */
    public String getMiddleName() {
        return middleName;
    }

    public String getId() {
        return id;
    }
}

File: lib/timetable/Person.rb
class Person
  include HappyMapper

  tag 'person'

  has_one :first_name, String, :tag => 'fn'
  has_one :middle_name, String, :tag => 'mn'
  has_one :last_name, String, :tag => 'ln'

  attribute :id, String

  def name
    result = last_name
    if first_name != '' and middle_name != ''
      result += ' ' + first_name[0] + '. ' + middle_name[0] + '.'
    end
    result
  end
end
